src/lib/featureProcessor.ts

```diff
@@ -1,81 +1,111 @@
 import { findAllFilesForPattern } from "./fileUtilities";
-import { fromPaths } from "gherkin";
+import { flatten, sortBy } from "lodash";
 import { io } from "cucumber-messages";
 import Wrapper = io.cucumber.messages.Wrapper;
-import IFeature = io.cucumber.messages.IFeature;
-
-const FEATURE_SORT_PATTERN =  ['name', 'scenario.name'];
+import { Feature } from "./testPlan";
+import path from "path";
 
 /**
  * Recursively traverses a directory to tranform all feature files into a sorted list of Gherkin Features
- *      Sort attributes: feature.name,scenario.name
+ *      sorted by the feature.name
  * @param directory Root directories to search recursively
  * @returns         List of Gherkin Features
  */
-export async function getFeaturesFromDirectory(
-  rootDirectory: string
-): Promise<IFeature[]> {
-  
-  const files = await findAllFeatureFiles(rootDirectory);
-  const messages = await parseFeatureFiles(files);
+export async function loadFeaturesFrom(
+   rootDirectory: string
+): Promise<Feature[]> {
+   const files = await findAllFeatureFiles(rootDirectory);
 
-  const errors = messages
-    //extract parsing errors
-    .filter(message => !!message.attachment && !!message.attachment.data)
-    //provide readable error message
-    .map(message => `File: ${message.attachment.source.uri}\n ${message.attachment.data}\n`);
-  
-  if  (errors.length > 0) {
-    throw new Error(`Gherkin Parsing Error\n${errors.join('\n')}`);
-  }
+   const messages: Wrapper[] = flatten(
+      await Promise.all(files.map(file => parseFeatureFile(file)))
+   );
 
-  const validFeatures = messages
-    //extract documents
-    .map(message => message.gherkinDocument)
-    //keep only valid features
-    .filter(document => !!document && !!document.feature)
-    //extract features
-    .map(document => document.feature);
-  
-    return _.orderBy(validFeatures, FEATURE_SORT_PATTERN);
-}
+   const errors = messages
+      // extract parsing errors
+      .filter(message => !!message.attachment && !!message.attachment.data)
+      .map(
+         message =>
+            `File: ${message.attachment.source.uri}\r\n ${
+               message.attachment.data
+            }\r\n`
+      );
+
+   if (errors.length > 0) {
+      throw new Error(`Gherkin Parsing Error\r\n${errors.join("\r\n")}`);
+   }
 
+   if (!messages.length) {
+      throw new Error(`Feature file is empty: in ${rootDirectory}`);
+   }
 
+   const validFeatures: Feature[] = messages
+      .map(message => message.gherkinDocument)
+      // keep only valid features
+      .filter(document => !!document && !!document.feature)
+      .map(document => {
+         return {
+            ...document.feature,
+            // Used to group features by their directory location
+            filename: formatFeatureName(document.uri),
+         };
+      });
+
+   return sortBy(validFeatures, "filename");
+}
+
+function formatFeatureName(name: string): string {
+   return name
```

Comment:
You might also consider a split and filter approach; something like:

```js
return name
  .replace(/\.feature$/, "") # Regex here insures that we only replace from the end of the string (paranoia)
  .split(/[\/\\]/)
  .filter((component) => ["features", "behavior", "manual", "test_data"].indexOf(component) < 0)
  .join(path.sep)
```

Reply:
That did not yield the same results (unit test failed).  I also re-wrote this in the next PR.  Can we ice this one and return to it?